lib/internal/Magento/Framework/MultiDimensionalIndexer/IndexStructureInterface.php

```diff
@@ -0,0 +1,44 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\Framework\MultiDimensionalIndexer;
+
+/**
+ * Represent manipulation with index structure
+ *
+ * @api
+ */
+interface IndexStructureInterface
+{
+    /**
+     * Create the Index Structure
+     *
+     * @param IndexName $indexName
+     * @param string $connectionName
+     * @throws \Magento\Framework\Exception\StateException
+     * @return void
+     */
+    public function create(IndexName $indexName, string $connectionName);
+
+    /**
+     * Delete the given Index
+     *
+     * @param IndexName $indexName
+     * @param string $connectionName
+     * @return void
+     */
+    public function delete(IndexName $indexName, string $connectionName);
+
+    /**
+     * Checks whether the Index exits
+     *
+     * @param IndexName $indexName
+     * @param string $connectionName
+     * @return bool
+     */
+    public function isExist(IndexName $indexName, string $connectionName): bool;
```

Comment:
Is there any obstacle in making `create` and `delete` idempotent eliminating this method?